presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoUpgrader.java

```diff
@@ -69,16 +74,57 @@ private static void upgradeUsingRpm(String pathToRpm, boolean checkDependencies)
             throws PrestoManagerException
     {
         checkRpmPackage(pathToRpm);
-        int upgradeRpm;
-        if (checkDependencies) {
-            upgradeRpm = executeCommand(SUBPROCESS_TIMEOUT, "sudo", "rpm", "-U", pathToRpm);
+        File configFiles = tarConfigFiles();
+        try {
+            int upgradeRpm;
+            if (checkDependencies) {
+                upgradeRpm = executeCommand(SUBPROCESS_TIMEOUT, "sudo", "rpm", "-U", pathToRpm);
+            }
+            else {
+                upgradeRpm = executeCommand(SUBPROCESS_TIMEOUT, "sudo", "rpm", "-U", "--nodeps", pathToRpm);
+            }
+            if (upgradeRpm != 0) {
+                throw new PrestoManagerException("Failed to upgrade presto", upgradeRpm);
+            }
+            deployConfigFiles(configFiles);
         }
-        else {
-            upgradeRpm = executeCommand(SUBPROCESS_TIMEOUT, "sudo", "rpm", "-U", "--nodeps", pathToRpm);
-        }
-        if (upgradeRpm != 0) {
-            throw new PrestoManagerException("Failed to upgrade presto", upgradeRpm);
+        finally {
+            if (!configFiles.delete()) {
+                LOGGER.warn("Failed to delete the tempFile: {}", configFiles.toString());
+            }
         }
         LOGGER.debug("Successfully upgraded presto");
     }
+
+    private static File tarConfigFiles()
+            throws PrestoManagerException
+    {
+        File tempFile;
+        try {
+            tempFile = File.createTempFile("ConfigFiles", ".tar.gz");
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to create temp file.", e);
+        }
+        int tarConfigFiles = executeCommand("tar", "-czv", "-f", tempFile.getAbsolutePath(), "-C", CONFIG_DIR.toString());
```

Comment:
`-czv` and `-f` can be combined into `-czvf`. Put the `"."` back at the end or it won't work.

I would still recommend `cp -r CONFIG_DIR tempDir`, though.